src/test/blockencodings_tests.cpp

```diff
@@ -302,6 +303,44 @@ BOOST_AUTO_TEST_CASE(EmptyBlockRoundTripTest)
     }
 }
 
+BOOST_AUTO_TEST_CASE(ReceiveWithExtraTransactions) {
+    CTxMemPool& pool = *Assert(m_node.mempool);
+    TestMemPoolEntryHelper entry;
+    CBlock block(BuildBlockTestCase());
+    std::vector<CTransactionRef> extra_txn;
+    extra_txn.resize(10);
+
+    CMutableTransaction mtx = BuildTransactionTestCase();
+    mtx.vin[0].prevout.hash = Txid::FromUint256(InsecureRand256());
+    mtx.vin[0].prevout.n = 0;
+    CTransactionRef non_block_tx = MakeTransactionRef(std::move(mtx));
+
+    LOCK2(cs_main, pool.cs);
+    pool.addUnchecked(entry.FromTx(block.vtx[2]));
+    BOOST_CHECK_EQUAL(pool.get(block.vtx[2]->GetHash()).use_count(), SHARED_TX_OFFSET + 0);
+
+    {
+        CBlockHeaderAndShortTxIDs cmpctblock{block};
+        PartiallyDownloadedBlock partialBlock(&pool);
+        PartiallyDownloadedBlock partialBlockWithExtra(&pool);
+
+        BOOST_CHECK(partialBlock.InitData(cmpctblock, extra_txn) == READ_STATUS_OK);
+        BOOST_CHECK( partialBlock.IsTxAvailable(0));
+        BOOST_CHECK(!partialBlock.IsTxAvailable(1));
+        BOOST_CHECK( partialBlock.IsTxAvailable(2));
+
+        // Add an unrelated tx to extra, and a tx from the block that's not in the mempool:
+        extra_txn[0] = non_block_tx;
```

Comment:
Should I 'assert something' about this `non_block_tx`? Like that it's not the same as the tx at `IsTxAvailable(1)`?

Reply:
> Should I 'assert something' about this non_block_tx? Like that it's not the same as the tx at IsTxAvailable(1) and that it's not in the block?

I think that's a good idea. Even though the name, comment and code make this obvious already, I think the assertion would (hopefully) prevent us from breaking the test in the future (e.g. someone modifies the test and adds the tx to the block for some reason).